alerting/src/main/kotlin/org/opensearch/alerting/model/suggestions/rules/util/SuggestionObjectType.kt

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.model.suggestions.rules.util
+
+import org.opensearch.alerting.model.Monitor
+import kotlin.reflect.KClass
+
+enum class SuggestionObjectType(val value: KClass<out Any>) {
+    MONITOR(Monitor::class);
+
+    companion object {
+        fun enumFromClass(type: KClass<*>) = SuggestionObjectType.values().first { it.value == type }
```

Comment:
So the original idea was that a check would be made to verify that the enum contains the value before making the call, otherwise, I throw an exception manually. The call to enumFromClass() does not have this kind of protection so I'll get on that. The call to enumFromString() does do this protection, in RestGetSuggestionsAction.kt line 58, but I'll put it in a try catch too. Could you tell me more about where null values might pop up?